File: 1680.concatenation-of-consecutive-binary-numbers/solution.go
func concatenatedBinary(n int) int {
    res := 0
    l := 0
    for i := 1; i <= n; i++ {
        if i & (i - 1) == 0 {
            l++
        }
        res = ((res << l) + i) % 1_000_000_007
    }
    return res
}

File: 1680.concatenation-of-consecutive-binary-numbers/solution.kt
const val MOD = 1_000_000_007

class Solution {
    fun concatenatedBinary(n: Int): Int {
        var res = 0L
        var l = 0
        for (i in 1..n) {
            if (i and i - 1 == 0) {
                l++
            }
            res = ((res shl l) + i) % MOD
        }
        return res.toInt()
    }
}